jaeger_client/codecs.py

```diff
@@ -232,3 +233,54 @@ def extract(self, carrier):
         return SpanContext(trace_id=trace_id, span_id=span_id,
                            parent_id=parent_id, flags=flags,
                            baggage=None)
+
+
+def header_to_hex(header):
+    if not isinstance(header, (str, unicode)):
+        raise SpanContextCorruptedException(
+            'malformed trace context "%s", expected hex string' % header)
+    try:
+        cast_header = int(header, 16)
+    except ValueError:
+        raise SpanContextCorruptedException(
+            'malformed trace context "%s", expected hex string' % header)
+    return cast_header
+
+
+class B3Codec(Codec):
+    """
+    Support B3 header properties
+
+    """
+    def __init__(self):
+        self.trace_header = 'X-B3-TraceId'
+        self.span_header = 'X-B3-SpanId'
+        self.parent_span_header = 'X-B3-ParentSpanId'
+        self.sampled_header = 'X-B3-Sampled'
+        self.flags_header = 'X-B3-Flags'
+
+    def inject(self, span_context, carrier):
+        if not isinstance(carrier, dict):
+            raise InvalidCarrierException('carrier not a dictionary')
+        carrier[self.trace_header] = format(span_context.trace_id, 'x')
```

Comment:
are you sure `format(n, 'x')` is the right format for Zipkin?  I seem to recall it being rather pendantic about the exact length of the headers ["32 or 16 lower-hex characters"](https://github.com/openzipkin/b3-propagation#traceid-1).

Reply:
I'll add a size validation for those things

Reply:
hi, sorry if I wasn't clear - I was referring to the `inject` method. The `format` function will not do 0-padding, so you are not guaranteed to produce 16/32 hex strings.

In the reading side (header_to_hex) I would actually be in favor of allowing any length, just like Jaeger does.

Reply:
Totally my mistake, I got too focused the hex format on the openzipkin spec. I'll do the correct change

Reply:
Also, it seems the java refuses hex strings of the wrong length https://github.com/jaegertracing/jaeger-client-java/blob/17b7a2c448a2d6beacede6e07e134580a2250493/jaeger-core/src/main/java/com/uber/jaeger/propagation/HexCodec.java

Should I keep or remove the validation?

Reply:
it refuses strings >32 as far as I can tell, but not anything in between.